ansible_ai_connect/ai/api/views.py

```diff
@@ -245,15 +245,18 @@ def handle_exception(self, exc):
     def finalize_response(self, request, response, *args, **kwargs):
         response = super().finalize_response(request, response, *args, **kwargs)
 
-        try:
-            model_meta_data: MetaData = apps.get_app_config("ai").get_model_pipeline(MetaData)
-            user = request.user
-            org_id = hasattr(user, "organization") and user.organization and user.organization.id
-            self.event.modelName = self.event.modelName or model_meta_data.get_model_id(
-                request.user, org_id, self.req_model_id
-            )
-        except (WcaNoDefaultModelId, WcaModelIdNotFound, WcaSecretManagerError):
-            pass
+        if request.user.is_authenticated:
+            try:
+                model_meta_data: MetaData = apps.get_app_config("ai").get_model_pipeline(MetaData)
+                user = request.user
+                org_id = (
+                    hasattr(user, "organization") and user.organization and user.organization.id
+                )
+                self.event.modelName = self.event.modelName or model_meta_data.get_model_id(
+                    request.user, org_id, self.req_model_id
+                )
+            except (WcaNoDefaultModelId, WcaModelIdNotFound, WcaSecretManagerError):
+                pass
```

Comment:
I know that we were passing this part before but what about a debug log here just for indicating the error type and its message as a debug log.

Reply:
@mabulgu I will open a separate PR for that part. Thanks for your comment.

Reply:
not a big deal. feel free to skip it as we haven't been doing it already